### PR TITLE
Email validation for managed members should only fail if it does not match the domain set to a broker

### LINKED ISSUE #29460: Email validation for managed members should only fail if it does not match the domain set to a broker
### Description

Initially, we were forcing members to use an email domain that must match one of the domains set to an organization.

As we updated the user profile validator to skip this validation for unmanaged members (e.g.: existing realm users added as a member) as per #29023, we need to also skip validations for managed members if their respective identity provider does not have a domain set.

This should also make the validation consistent with the `OrganizationAuthenticator` where users are not automatically redirected if the identity provider does not have a domain set.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #29460

### CODE REVIEW

**Comment by alice-wondered** [comment]
LGTM. Lots of database transactions for each validation but I'm not sure that's avoidable; can also be addressed if there's an issue rather than trying to optimize for that immediately

File: services/src/main/java/org/keycloak/organization/validator/OrganizationMemberValidator.java [hunk 1]
```diff
@@ -73,54 +75,81 @@ public boolean isSupported(Scope config) {
...
+        if (!UserModel.EMAIL.equals(inputHint)) {
+            return;
+        }
+
+        if (StringUtil.isBlank(email)) {
+            context.addError(new ValidationError(ID, inputHint, "Email not set"));
+            return;
+        }
+
+        if (!emailValidator().validate(email, inputHint, context).isValid()) {
```
Comment (sguilhen) [inline services/src/main/java/org/keycloak/organization/validator/OrganizationMemberValidator.java:87]: I think I missed this when we introduced this validator, but shouldn't this situation add an error to the context?

Comment (pedroigor) [reply services/src/main/java/org/keycloak/organization/validator/OrganizationMemberValidator.java:87]: It will add one. See `EmailValidator`.

**Comment by pedroigor** [comment]
> LGTM. Lots of database transactions for each validation but I'm not sure that's avoidable; can also be addressed if there's an issue rather than trying to optimize for that immediately

Wouldn't be addressed by caching? Or do you think we can improve the logic a bit to avoid any unnecessary call?

**Review by sguilhen — APPROVED** [review]

**Comment by alice-wondered** [comment]
> @alice-wondered Wouldn't be addressed by caching? Or do you think we can improve the logic a bit to avoid any unnecessary call?

Yeah I think caching will account for the main concern I have for sure!

**Review by ahus1 — APPROVED** [review]
Approving based on @sguilhen's review.

**Pull request merged**